src/mail-app/search/view/SearchViewModel.ts

```diff
@@ -524,10 +524,20 @@ export class SearchViewModel {
 			this.freeToAskAboutExtendingIndex = true
 			if (confirmed) {
 				this._startDate = startDate
-				this.indexerFacade.extendMailIndex(startDate.getTime()).then(() => {
-					this.updateSearchUrl()
-					this.updateUi()
+
+				const listModel = this._listModel
+				const endDate = this._endDate
+
+				this.indexerFacade.extendMailIndex(startDate.getTime()).then(async () => {
+					// don't do anything further if the search parameters were changed
+					if (this._startDate !== startDate || this._endDate !== endDate || this._listModel !== listModel || !this.freeToAskAboutExtendingIndex) {
```

Comment:
is it enough to compare the times only? we usually use `isSameSearchRestriction()`